File: VB/asp-net-core-server/Models/NorthwindContext.vb
Imports Microsoft.EntityFrameworkCore

Namespace AspNetCoreDashboardBackend

    Public Partial Class NorthwindContext
        Inherits DbContext

        Public Sub New(ByVal options As DbContextOptions(Of NorthwindContext))
            MyBase.New(options)
        End Sub

        Public Overridable Property Products As DbSet(Of Product)

        Protected Overrides Sub OnModelCreating(ByVal modelBuilder As ModelBuilder)
            modelBuilder.Entity(Of Product)().[Property](Function(e) e.ProductID).HasConversion(Of String)()
        End Sub
    End Class
End Namespace

File: asp-net-core-server/Models/NorthwindContext.cs
using Microsoft.EntityFrameworkCore;

namespace AspNetCoreDashboardBackend {
    public partial class NorthwindContext : DbContext {
        public NorthwindContext(DbContextOptions<NorthwindContext> options)
            : base(options) {

        }

        public virtual DbSet<Product> Products { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder) {
            modelBuilder
                .Entity<Product>()
                .Property(e => e.ProductID)
                .HasConversion<string>();
        }
    }
}